merlin/analysis/segment.py

```diff
@@ -168,32 +168,40 @@ def _construct_fov_tree(self, tiledPositions: pandas.DataFrame,
     def _intial_clean(self, currentFOV: int):
         currentCells = self.segmentTask.get_feature_database()\
             .read_features(currentFOV)
-        return [cell for cell in currentCells 
+        return [cell for cell in currentCells
                 if len(cell.get_bounding_box()) == 4 and cell.get_volume() > 0]
 
-    def _secondary_assignments(self, currentFOV: int,
-                               secondaryAssignmentDict: Dict):
-        currentCells = self._intial_clean(currentFOV)
-
-        secondaryAssignments = [
-            [x, secondaryAssignmentDict[currentFOV][x.get_feature_id()]]
-            for x in currentCells
-            if x.get_feature_id()
-            in secondaryAssignmentDict[currentFOV]]
-        return secondaryAssignments
-
     def _append_cells_to_spatial_tree(self, tree: rtree.index.Index,
                                       cells: List, idToNum: Dict):
         for element in cells:
             tree.insert(idToNum[element.get_feature_id()],
-                        element.get_bounding_box(), obj=element.get_fov())
-
-    def _run_analysis(self, fragmentIndex):
+                        element.get_bounding_box(), obj=element)
+
+    def _return_overlapping_cells(self, currentCell, cells: List):
+        areas = [currentCell.intersection(x) for x in cells]
+        overlapping = [cells[i] for i, x in enumerate(areas) if x > 0]
+        benchmark = currentCell.intersection(currentCell)
+        contained = [x for x in overlapping if
+                     x.intersection(currentCell) == benchmark]
+        if len(contained) > 1:
+            overlapping = []
```

Comment:
Can you move the functionality of this function and the remove_overlapping_cells function into a utility class and write unit tests to test the edge cases so we can be sure this works as expected this time? This could possibly fit into the util/spatialfeature.py module.